apps/platform/src/components/dashboard/variable/addVariableDialogue/index.tsx

```diff
@@ -28,78 +21,104 @@ import {
 } from '@/store'
 import { useHttp } from '@/hooks/use-http'
 
+interface VariableFormState {
+  variableName: string
+  note: string
+  environmentEntries: Record<string, string>
+}
+
 export default function AddVariableDialogue() {
-  const [isCreateVariableOpen, setIsCreateVariableOpen] = useAtom(
-    createVariableOpenAtom
-  )
+  const [isCreateVariableOpen, setIsCreateVariableOpen] = useAtom(createVariableOpenAtom)
   const environments = useAtomValue(environmentsOfProjectAtom)
   const selectedProject = useAtomValue(selectedProjectAtom)
   const setVariables = useSetAtom(variablesOfProjectAtom)
 
-  const [newVariableData, setNewVariableData] = useState({
+  const [newVariableData, setNewVariableData] = useState<VariableFormState>({
     variableName: '',
     note: '',
-    environmentSlug: environments[0]?.slug,
-    environmentValue: ''
+    environmentEntries: {}
   })
+
   const [isLoading, setIsLoading] = useState<boolean>(false)
 
-  const createVariable = useHttp(() =>
-    ControllerInstance.getInstance().variableController.createVariable({
-      name: newVariableData.variableName,
-      projectSlug: selectedProject!.slug,
-      entries: newVariableData.environmentValue
-        ? [
-            {
-              value: newVariableData.environmentValue,
-              environmentSlug: newVariableData.environmentSlug
-            }
-          ]
-        : undefined,
-      note: newVariableData.note
+  // Sync environment entries when environments change
```

Comment:
this block is irrelevant since new variable data is only used when creating a variable. it is bound to change on page reload / component reload